java/flight/flight-sql/src/test/java/org/apache/arrow/flight/sql/example/FlightSqlExample.java

```diff
@@ -429,6 +453,80 @@ private static VectorSchemaRoot getRoot(final ResultSet data, final BufferAlloca
     return new VectorSchemaRoot(singletonList(dataVector));
   }
 
+  private static VectorSchemaRoot getTypeInfoRoot(CommandGetTypeInfo request, ResultSet typeInfo,
+                                                  final BufferAllocator allocator)
+      throws SQLException {
+    Objects.requireNonNull(allocator, "BufferAllocator cannot be null.");
+
+    VectorSchemaRoot root = VectorSchemaRoot.create(Schemas.GET_TYPE_INFO_SCHEMA, allocator);
+
+    VarCharVector typeNameVector = (VarCharVector) root.getVector("type_name");
+    IntVector dataTypeVector = (IntVector) root.getVector("data_type");
+    IntVector columnSizeVector = (IntVector) root.getVector("column_size");
+    VarCharVector literalPrefixVector = (VarCharVector) root.getVector("literal_prefix");
+    VarCharVector literalSuffixVector = (VarCharVector) root.getVector("literal_suffix");
+    VarCharVector createParamsVector = (VarCharVector) root.getVector("create_params");
+    IntVector nullableVector = (IntVector) root.getVector("nullable");
+    BitVector caseSensitiveVector = (BitVector) root.getVector("case_sensitive");
+    IntVector searchableVector = (IntVector) root.getVector("searchable");
+    BitVector unsignedAttributeVector = (BitVector) root.getVector("unsigned_attribute");
+    BitVector fixedPrecScaleVector = (BitVector) root.getVector("fixed_prec_scale");
+    BitVector autoIncrementVector = (BitVector) root.getVector("auto_increment");
+    VarCharVector localTypeNameVector = (VarCharVector) root.getVector("local_type_name");
+    IntVector minimumScaleVector = (IntVector) root.getVector("minimum_scale");
+    IntVector maximumScaleVector = (IntVector) root.getVector("maximum_scale");
+    IntVector sqlDataTypeVector = (IntVector) root.getVector("sql_data_type");
+    IntVector sqlDatetimeSubVector = (IntVector) root.getVector("sql_datetime_sub");
+    IntVector numPrecRadixVector = (IntVector) root.getVector("num_prec_radix");
```

Comment:
Since the types don't matter, why not inline these into the `mapper.put` calls below?

Reply:
Fixed